crypto/dsa/dsa_ossl.c

```diff
@@ -189,17 +193,14 @@ static int dsa_sign_setup(DSA *dsa, BN_CTX *ctx_in,
 
     /*
      * We do not want timing information to leak the length of k, so we
-     * compute g^k using an equivalent exponent of fixed length. (This
-     * is a kludge that we need because the BN_mod_exp_mont() does not
-     * let us specify the desired timing behaviour.)
+     * compute ensure the k is longer than the q value.  Any code that checks
+     * for this an does a modulo redution will consequently also be consant
+     * time.  This is a kludge that we need because the BN_mod_exp_mont()
+     * does not let us specify the desired timing behaviour.
      */
 
-    if (!BN_add(k, k, dsa->q))
+    if (!BN_set_bit(k, BN_num_bits(dsa->q) + 1))
```

Comment:
BTW, just in case, on side note, +1 means that it's *two* bits wider than q. I mean -1 means same width, 0 - one bit wider, +1 - two bits wider.

Reply:
So I should switch to 0 then?

Reply:
It doesn't really matter from mod_exp viewpoint, as it will take same amount of iterations. Other than that I'm kind of groggy to think about it, too late here... In fact signing off for today, sorry...